src/structures/TikTokClient.js

```diff
@@ -5,7 +5,7 @@ class TikTokClient {
   async getPost(match) {
     const url = match[0];
     // This should be safe as our regexes earlier prevent any weirdness
-    return await sh(`yt-dlp "${url.replace(/"/g, '\"')}" -j`).then((stdout)=>{
```

Comment:
the .replace needs to be kept in, if the URL vm.tiktok.com gets redirected to contains a '"', this breaks the command.

Reply:
I should also add a check for ! and $ becuase those are special chars as well

Reply:
The reason why I ditch the .replace is because it crashes the whole bot (at least on my setup NodeJS 17.8.0 with RE_EMBED)
This is the error log with added url printout 
```
info index Ready! logged in as DEEZEmbed#2440(957830877331742750)
URL {
  href: 'https://www.tiktok.com/@syafiqalzro/video/7071849663330045211?_d=secCgwIARCbDRjEFSADKAESPgo8lV%2FGLsvCX77vF3SZBepT4h4Qm6XJcO5vv%2FH%2FSshUrpypCg%2FI2yHzK2gUeg8BeawgBVISapr%2F%2B6EldReBGgA%3D&_r=1&language=en&mid=7049007051422321434&preview_pb=0&region=ID&sec_user_id=MS4wLjABAAAA-TsWqfpHIAW-jayCB6rDgT0g5bao3lATEWLYguQrfXp4JzrxKR5Ua0KW14z38E7p&share_app_id=1180&share_item_id=7071849663330045211&share_link_id=df4a174d-ab77-4c95-a97a-7e34ca1fbfed&social_sharing=v4&source=h5_t&timestamp=1646720537&u_code=dgfd81l88gaml5&user_id=6917951947374887937&utm_campaign=client_share&utm_medium=android&utm_source=copy',
  origin: 'https://www.tiktok.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'www.tiktok.com',
  hostname: 'www.tiktok.com',
  port: '',
  pathname: '/@syafiqalzro/video/7071849663330045211',
  search: '?_d=secCgwIARCbDRjEFSADKAESPgo8lV%2FGLsvCX77vF3SZBepT4h4Qm6XJcO5vv%2FH%2FSshUrpypCg%2FI2yHzK2gUeg8BeawgBVISapr%2F%2B6EldReBGgA%3D&_r=1&language=en&mid=7049007051422321434&preview_pb=0&region=ID&sec_user_id=MS4wLjABAAAA-TsWqfpHIAW-jayCB6rDgT0g5bao3lATEWLYguQrfXp4JzrxKR5Ua0KW14z38E7p&share_app_id=1180&share_item_id=7071849663330045211&share_link_id=df4a174d-ab77-4c95-a97a-7e34ca1fbfed&social_sharing=v4&source=h5_t&timestamp=1646720537&u_code=dgfd81l88gaml5&user_id=6917951947374887937&utm_campaign=client_share&utm_medium=android&utm_source=copy',
  searchParams: URLSearchParams {
    '_d' => 'secCgwIARCbDRjEFSADKAESPgo8lV/GLsvCX77vF3SZBepT4h4Qm6XJcO5vv/H/SshUrpypCg/I2yHzK2gUeg8BeawgBVISapr/+6EldReBGgA=',
    '_r' => '1',
    'language' => 'en',
    'mid' => '7049007051422321434',
    'preview_pb' => '0',
    'region' => 'ID',
    'sec_user_id' => 'MS4wLjABAAAA-TsWqfpHIAW-jayCB6rDgT0g5bao3lATEWLYguQrfXp4JzrxKR5Ua0KW14z38E7p',
    'share_app_id' => '1180',
    'share_item_id' => '7071849663330045211',
    'share_link_id' => 'df4a174d-ab77-4c95-a97a-7e34ca1fbfed',
    'social_sharing' => 'v4',
    'source' => 'h5_t',
    'timestamp' => '1646720537',
    'u_code' => 'dgfd81l88gaml5',
    'user_id' => '6917951947374887937',
    'utm_campaign' => 'client_share',
    'utm_medium' => 'android',
    'utm_source' => 'copy' },
  hash: ''
}
C:\git\discord-twitter-video-embeds\src\structures\TikTokClient.js:9
    return await sh(`yt-dlp '${url.replace(/'/g, '\'\\\'\'')}' -j`).then((stdout)=>{
                                   ^

TypeError: url.replace is not a function
    at TikTokClient.getPost (C:\git\discord-twitter-video-embeds\src\structures\TikTokClient.js:9:36)
    at C:\git\discord-twitter-video-embeds\src\structures\TikTokRedirect.js:17:29
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getPost (C:\git\discord-twitter-video-embeds\src\util\getPosts.js:42:12)
    at async Promise.all (index 0)
    at async Client.handleMessage (C:\git\discord-twitter-video-embeds\src\events\messageCreate.js:126:17)
```
Also on the new commit, you might want to replace the `''` encapsulation with `""` as (at least on windows batch) the url with `&` special chars will broke the script 
```
node:internal/errors:844
  const err = new Error(message);
              ^

Error: Command failed: yt-dlp 'https://www.tiktok.com/@syafiqalzro/video/7071849663330045211?_d=secCgwIARCbDRjEFSADKAESPgo8lV%2FGLsvCX77vF3SZBepT4h4Qm6XJcO5vv%2FH%2FSshUrpypCg%2FI2yHzK2gUeg8BeawgBVISapr%2F%2B6EldReBGgA%3D&_r=1&language=en&mid=7049007051422321434&preview_pb=0&region=ID&sec_user_id=MS4wLjABAAAA-TsWqfpHIAW-jayCB6rDgT0g5bao3lATEWLYguQrfXp4JzrxKR5Ua0KW14z38E7p&share_app_id=1180&share_item_id=7071849663330045211&share_link_id=df4a174d-ab77-4c95-a97a-7e34ca1fbfed&social_sharing=v4&source=h5_t&timestamp=1646720537&u_code=dgfd81l88gaml5&user_id=6917951947374887937&utm_campaign=client_share&utm_medium=android&utm_source=copy' -j
ERROR: [generic] "'https://www.tiktok.com/@syafiqalzro/video/7071849663330045211?_d=secCgwIARCbDRjEFSADKAESPgo8lV%2FGLsvCX77vF3SZBepT4h4Qm6XJcO5vv%2FH%2FSshUrpypCg%2FI2yHzK2gUeg8BeawgBVISapr%2F%2B6EldReBGgA%3D" is not a valid URL. Set --default-search "ytsearch" (or run  yt-dlp "ytsearch:'https://www.tiktok.com/@syafiqalzro/video/7071849663330045211?_d=secCgwIARCbDRjEFSADKAESPgo8lV%2FGLsvCX77vF3SZBepT4h4Qm6XJcO5vv%2FH%2FSshUrpypCg%2FI2yHzK2gUeg8BeawgBVISapr%2F%2B6EldReBGgA%3D" ) to search YouTube
'_r' is not recognized as an internal or external command,
operable program or batch file.
'language' is not recognized as an internal or external command,
operable program or batch file.
'mid' is not recognized as an internal or external command,
operable program or batch file.
'preview_pb' is not recognized as an internal or external command,
operable program or batch file.
'region' is not recognized as an internal or external command,
operable program or batch file.
'sec_user_id' is not recognized as an internal or external command,
operable program or batch file.
'share_app_id' is not recognized as an internal or external command,
operable program or batch file.
'share_item_id' is not recognized as an internal or external command,
operable program or batch file.
'share_link_id' is not recognized as an internal or external command,
operable program or batch file.
'social_sharing' is not recognized as an internal or external command,
operable program or batch file.
'source' is not recognized as an internal or external command,
operable program or batch file.
'timestamp' is not recognized as an internal or external command,
operable program or batch file.
'u_code' is not recognized as an internal or external command,
operable program or batch file.
'user_id' is not recognized as an internal or external command,
operable program or batch file.
'utm_campaign' is not recognized as an internal or external command,
operable program or batch file.
'utm_medium' is not recognized as an internal or external command,
operable program or batch file.
'utm_source' is not recognized as an internal or external command,
operable program or batch file.

    at ChildProcess.exithandler (node:child_process:398:12)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1090:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: "yt-dlp 'https://www.tiktok.com/@syafiqalzro/video/7071849663330045211?_d=secCgwIARCbDRjEFSADKAESPgo8lV%2FGLsvCX77vF3SZBepT4h4Qm6XJcO5vv%2FH%2FSshUrpypCg%2FI2yHzK2gUeg8BeawgBVISapr%2F%2B6EldReBGgA%3D&_r=1&language=en&mid=7049007051422321434&preview_pb=0&region=ID&sec_user_id=MS4wLjABAAAA-TsWqfpHIAW-jayCB6rDgT0g5bao3lATEWLYguQrfXp4JzrxKR5Ua0KW14z38E7p&share_app_id=1180&share_item_id=7071849663330045211&share_link_id=df4a174d-ab77-4c95-a97a-7e34ca1fbfed&social_sharing=v4&source=h5_t&timestamp=1646720537&u_code=dgfd81l88gaml5&user_id=6917951947374887937&utm_campaign=client_share&utm_medium=android&utm_source=copy' -j"
}
```